File: DataCore/XInstructor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace x_srv.models { 
	/* XInstructor -  Инструктор */ 

 public class  XInstructorInfo { // Описание
	 public System.Guid  XInstructorInfoId{ get; set; } // Идентификатор (первичный ключ)
	 public List<XInstructorStatus>  XInstructorStatus { get; set; } // дочерний раздел: Статусы
	[Required]
	public string  family{ get; set; } // Фамилия
	[Required]
	public string  name{ get; set; } // Имя
	[Required]
	public string  middleName{ get; set; } // Отчество
	public string  eMail{ get; set; } // e-mail
	public string  phone{ get; set; } // Телефон
	public string  photoUrl{ get; set; } // Фотография
	public string  localPhone{ get; set; } // Местный телефон
 }

 public class  XInstructorStatus { // Статусы
		 public System.Guid  XInstructorStatusId{ get; set; } // Идентификатор (первичный ключ)
		[Required]
		 public System.Guid  XInstructorInfoId { get; set; } // обратная ссылка на: Описание
		public System.Guid  theStatus { get; set; } //Статус
		[ForeignKey("theStatus")]
		public XStatus XStatus { get; set; } // Объект - Статус
	public int?  sequence{ get; set; } // Порядок
	}
}

File: client/src/app/XInstructor.ts
import { enums } from './enums';

export namespace XInstructor { 
	/* XInstructor -  Инструктор */ 

 export interface   xInstructorInfo { // Описание
	xInstructorInfoId:string; // Primary key field
	family:string; // Фамилия
	name:string; // Имя
	middleName:string; // Отчество
	eMail:string; // e-mail
	phone:string; // Телефон
	photoUrl:string; // Фотография
	localPhone:string; // Местный телефон
 }

 export interface   xInstructorStatus { // Статусы
	xInstructorStatusId:string; // Primary key field
	  xInstructorInfoId:string; // Описание
	theStatus:string; //Статус -> XStatus
	sequence:Number; // Порядок
	// add dereference fields 
	theStatus_name :string; // dereference for XStatus
 }
}